### ISSUE TITLE
Add support for workspaces

### ISSUE THREAD

**Issue by mheap** [Kong/terraform-provider-kong-gateway#2]
Rough idea: 

* Require a workspace name in the URL in OAS generation
* Remove `/default` workspace in a hook to make the provider work with OSS and EE
* Make `workspace` param optional for all entities

**Comment by samjsharpe** [comment]
Hi @mheap do you have any futher plans to do this? We're using the philip-labs terraform provider and we'd like to move to the Kong "official" provider, but we need workspace support.

**Comment by mheap** [comment]
@samjsharpe Yep! Still planned. This provider is still in beta, and we had a new engineer start yesterday who will be focusing on the Terraform providers. Watch this space
